core/dbt/task/debug.py

```diff
@@ -182,10 +182,10 @@ def _choose_profile_names(self) -> Optional[List[str]]:
         project_profile: Optional[str] = None
         if os.path.exists(self.project_path):
             try:
-                project_profile = load_yaml_text(
-                    dbt.clients.system.load_file_contents(self.project_path)
-                ).get('profile')
-            except dbt.exceptions.Exception:
+                project_profile = Project.partial_load(
```

Comment:
I think calling `dbt.config.project._raw_project_from()` would be more appropriate here, but I've avoided that because of the convention that objects named with a leading underscore should not be accessed outside of their module or class.